model/task_lifecycle.go

```diff
@@ -217,12 +214,20 @@ func DeactivatePreviousTasks(taskId, caller string) error {
 	if err != nil {
 		return err
 	}
+	displayNames := []string{t.DisplayName}
+	for _, et := range t.ExecutionTasks {
```

Comment:
for performance reasons we should try to make this an `$in` query rather than a bunch of findOnes

Reply:
fixed